### PR TITLE
Add messages table and index

### PR DESCRIPTION
**Pull request by carlad**
This addresses this issue: https://github.com/travis-pro/team-teal/issues/2174 which is part of the travis-yml project, although I believe this table is required for other message reporting too.

Migrations have been run locally without error.

### CODE REVIEW

**Comment by svenfuchs** [comment]
From Slack:

> sven
> i discussed the idea of having a more generic concept of “messages” with @joe the other day. i’m not sure where we are with that, and if we need to run it by anyone else. but the idea would be to have a table `messages` that is polymorphic to requests, builds, stages, jobs … and possibly other stuff. what do you think?

> the question would be what a good name is for the association

> what’s the thing that a message belongs to

> `object_id` and `object_type` … except that `object_id` is taken in ruby, pretty sure we shouldn’t use that

> joe
> `subject_id` `subject_type`

Context is that we've had lots of situations in the past where we'd want to communicate something about a request, build, or job to the user, but could not because we don't have a concept for that in our domain model. Instead we have then resorted to such things as creating, but immediately canceling a build, only to write the respective message to the build log. 

This whole train of thought has now come up again when I have been working on conditional stages and jobs. With the current implementation Gatekeeper would just silently filter stages and jobs without letting the user know, because there's no good place for storing such a message.

With a generic messages concept we could just serve whatever messages belong to an object such as a request, build, stage, or job on the API and let clients decide what to do with it (e.g. where to display them in the UI).

If we decided to do this the table would look like:

```ruby
create_table "request_messages", force: :cascade do |t|
  t.integer "subject_id"
  t.integer "subject_type"
  t.string "level"
  t.string "key"
  t.string "code"
  t.jsonb "args"
  t.datetime "created_at", null: false
  t.datetime "updated_at", null: false
end
```

(I'm not sure what `force: :cascade` does, but I think it's good practice to define associations at the top.)

**Comment by carlad** [comment]
http://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-create_table
```
:force
Set to true to drop the table before creating it. Set to :cascade to drop dependent objects as well. Defaults to false.
```

**Comment by BanzaiMan** [comment]
The column names seem non-descriptive. What do `level`, `key`, `code`, `args` mean?

**Comment by svenfuchs** [comment]
@BanzaiMan good call :)

* `level` is a severity level (like log levels, travis-yml produces info, warning, error level messages)
* `key` is a fully qualified key that refers to a node in a hierarchy (e.g. `matrix.include.deploy.provider`)
* `code` is a message type (e.g. `build_config. invalid_type` would refer to a missing key in `.travis.yml`)
* `args` provides extra information about the message, or parametrizes the message (e.g.  `type_expected: :map, type_given: :string`)

Does that make sense? I think all of these are pretty generic, except maybe for the `key` which probably only is useful for the build config parsing messages.

**Comment by carlad** [comment]
Just checking on whether `subject_type` is an integer @svenfuchs , or a string?

**Comment by joecorcoran** [comment]
@carlad So the intention with `subject` is that it would be a polymorphic relationship. This means that a request, a build, a job, whatever, can have the same relationship with a message, without needing specific fields.

When ActiveRecord stores a polymorphic relationship, it needs to use two fields instead of the usual one, because it needs not only the id of the record, but the name of the class as well.

```
+------------+--------------+
| subject_id | subject_type |
+------------+--------------+
|          1 | Build        |
|          2 | Job          |
|         99 | Request      |
+------------+--------------+
```

So `subject_type` is a string, and `subject_id` is an integer. You wouldn't need the specific relationships declared as you wrote above.

In general I like this idea. I think there is a good argument to be made for having messages that can be attached to requests/builds/jobs, namely that we can display info/warnings/errors in the UI, in the specific locations that they belong. I understand @igorwwwwwwwwwwwwwwwwwwww had some reservations about it being too entangled. I think we should think about it a little longer before adding it.

What's not 100% clear to me:

1. How the key field fits into the generic concept. It feels a bit too strongly related to config parsing to me, as @svenfuchs mentions above.
2. What it looks like when API returns these records, and how much work is done to construct the human-readable messages server side vs. client side.

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ services:
   - postgresql
 
 addons:
-  postgresql: "9.3"
+  postgresql: "9.4"
```
Comment (joecorcoran) [inline .travis.yml]: Did we move production up to 9.4?

Comment (carlad) [reply .travis.yml]: yes

Comment (carlad) [reply .travis.yml]: to accommodate jsonb column types

Comment (carlad) [reply .travis.yml]: Ah - no - we did not. 
The database for `travis-yml-checker` uses 9.4 in order to accommodate `jsonb` column types.

Let's do an update next week then...

**Comment by carlad** [comment]
So we've reverted to a `json` column type in order to be backwards compatible with Enterprise, who have raised a ticket to upgrade their version of postgresql to one compatible with `jsonb` https://github.com/travis-pro/travis-enterprise/issues/197

**Pull request merged**